File: app/src/app/models/ExamInstance.ts
import { ExamQuestion } from "./ExamQuestion";
import { GeneratedLink } from "./GeneratedLink";

export interface ExamInstance {
    examInstanceId: string;
    examDefinitionId: string;
    startedTime: string;
    endTime: string;
    duration: string;
    completionTime: string;
    generatedLink: GeneratedLink;
    createdBy: string;
    takenBy: string;
    status: string;
    questions: ExamQuestion;

}

File: exam-microservice/src/main/java/com/example/exammicroservice/model/ExamInstance.java
package com.example.exammicroservice.model;

import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "ExamInstance")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ExamInstance {
    @Id
    String examInstanceId;
    String examDefinitionId;
    String startedTime;// take exam
    String endTime;// submit
    String duration;// check if user exceeded duration(calculated)
    String completionTime;// submit exam calsulated
    GeneratedLink generatedLink;//
    String createdBy;//
    String takenBy;
    StatusEnum status;// take exam set initially null or assigned
    List<ExamQuestion> examQuestions;// take exam

}
